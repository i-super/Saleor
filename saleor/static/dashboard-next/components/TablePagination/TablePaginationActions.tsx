import IconButton from "material-ui/IconButton";
import KeyboardArrowLeft from "material-ui/internal/svg-icons/KeyboardArrowLeft";
import KeyboardArrowRight from "material-ui/internal/svg-icons/KeyboardArrowRight";
import { withStyles } from "material-ui/styles";
import * as React from "react";

const decorate = withStyles(
  theme => ({
    root: {
      color: theme.palette.text.secondary,
      flexShrink: 0,
      marginLeft: theme.spacing.unit * 2.5
    }
  }),
  {
    name: "MuiTablePaginationActions",
    withTheme: true
  }
);

interface TablePaginationActionsProps {
  backIconButtonProps: any;
  classes: any;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
  nextIconButtonProps: any;
  onNextPage(event);
  onPreviousPage(event);
}

const TablePaginationActions = decorate<TablePaginationActionsProps>(
  ({
    backIconButtonProps,
    classes,
    hasNextPage,
    hasPreviousPage,
    nextIconButtonProps,
    onNextPage,
    onPreviousPage,
    theme,
    ...other
  }) => (
    <div className={classes.root} {...other}>
      <IconButton
        onClick={onPreviousPage}
        disabled={!hasPreviousPage}
        {...backIconButtonProps}
      >
        {theme.direction === "rtl" ? (
          <KeyboardArrowRight />
        ) : (
          <KeyboardArrowLeft />
        )}
      </IconButton>
      <IconButton
        onClick={onNextPage}
        disabled={!hasNextPage}
        {...nextIconButtonProps}
      >
        {theme.direction === "rtl" ? (
          <KeyboardArrowLeft />
        ) : (
          <KeyboardArrowRight />
        )}
      </IconButton>
    </div>
  )
);

export default TablePaginationActions;
