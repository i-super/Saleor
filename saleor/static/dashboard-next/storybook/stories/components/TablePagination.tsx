import { storiesOf } from "@storybook/react";
import Table from "material-ui/Table";
import * as React from "react";

import TablePagination from "../../../components/TablePagination";

storiesOf("Generics / TablePagination", module)
  .add("no previous / next page", () => (
    <Table>
      <TablePagination
        colSpan={1}
        hasNextPage={false}
        hasPreviousPage={false}
        onPreviousPage={() => {}}
        onNextPage={() => {}}
      />
    </Table>
  ))
  .add("previous page available", () => (
    <Table>
      <TablePagination
        colSpan={1}
        hasNextPage={false}
        hasPreviousPage={true}
        onPreviousPage={() => {}}
        onNextPage={() => {}}
      />
    </Table>
  ))
  .add("next page available", () => (
    <Table>
      <TablePagination
        colSpan={1}
        hasNextPage={true}
        hasPreviousPage={false}
        onPreviousPage={() => {}}
        onNextPage={() => {}}
      />
    </Table>
  ))
  .add("both previous and next pages are available", () => (
    <Table>
      <TablePagination
        colSpan={1}
        hasNextPage={true}
        hasPreviousPage={true}
        onPreviousPage={() => {}}
        onNextPage={() => {}}
      />
    </Table>
  ));
