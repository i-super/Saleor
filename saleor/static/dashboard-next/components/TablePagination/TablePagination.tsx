// @inheritedComponent TableCell

import { withStyles } from "@material-ui/core/styles";
import TableCell from "@material-ui/core/TableCell";
import Toolbar from "@material-ui/core/Toolbar";
import * as React from "react";

import TablePaginationActions from "./TablePaginationActions";

const decorate = withStyles(
  theme => ({
    actions: {
      color: theme.palette.text.secondary,
      flexShrink: 0,
      marginLeft: theme.spacing.unit * 2.5
    },
    caption: {
      flexShrink: 0
    },
    input: {
      flexShrink: 0,
      fontSize: "inherit"
    },
    root: {
      "&:last-child": {
        padding: 0
      }
    },
    select: {
      paddingLeft: theme.spacing.unit,
      paddingRight: theme.spacing.unit * 2
    },
    selectIcon: {
      top: 1
    },
    selectRoot: {
      color: theme.palette.text.secondary,
      marginLeft: theme.spacing.unit,
      marginRight: theme.spacing.unit * 4
    },
    spacer: {
      flex: "1 1 100%"
    },
    toolbar: {
      height: 56,
      minHeight: 56,
      paddingRight: 2
    }
  }),
  {
    name: "TablePagination"
  }
);

interface TablePaginationProps {
  Actions?: string | React.ComponentType;
  backIconButtonProps?: any;
  colSpan: number;
  component?: string | React.ComponentType;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
  nextIconButtonProps?: any;
  onNextPage(event);
  onPreviousPage(event);
}

const TablePagination = decorate<TablePaginationProps>(
  ({
    Actions,
    backIconButtonProps,
    classes,
    colSpan: colSpanProp,
    component: Component,
    hasNextPage,
    hasPreviousPage,
    nextIconButtonProps,
    onNextPage,
    onPreviousPage,
    ...other
  }) => {
    let colSpan;

    if (Component === TableCell || Component === "td") {
      colSpan = colSpanProp || 1000;
    }

    return (
      <Component className={classes.root} colSpan={colSpan} {...other}>
        <Toolbar className={classes.toolbar}>
          <div className={classes.spacer} />
          <Actions
            backIconButtonProps={backIconButtonProps}
            hasNextPage={hasNextPage}
            hasPreviousPage={hasPreviousPage}
            nextIconButtonProps={nextIconButtonProps}
            onNextPage={onNextPage}
            onPreviousPage={onPreviousPage}
          />
        </Toolbar>
      </Component>
    );
  }
);
TablePagination.defaultProps = {
  Actions: TablePaginationActions,
  component: TableCell
};

TablePagination.displayName = "TablePagination";
export default TablePagination;
