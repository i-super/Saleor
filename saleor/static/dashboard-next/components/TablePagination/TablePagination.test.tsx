import Table from "material-ui/Table";
import * as React from "react";
import * as renderer from "react-test-renderer";

import TablePagination from "./TablePagination";

describe("<TablePagination />", () => {
  it("renders when no previous / next page are available", () => {
    const component = renderer.create(
      <Table>
        <TablePagination
          colSpan={1}
          hasNextPage={false}
          hasPreviousPage={false}
          onPreviousPage={jest.fn()}
          onNextPage={jest.fn()}
        />
      </Table>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when previous page is available", () => {
    const component = renderer.create(
      <Table>
        <TablePagination
          colSpan={1}
          hasNextPage={false}
          hasPreviousPage={true}
          onPreviousPage={jest.fn()}
          onNextPage={jest.fn()}
        />
      </Table>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when next page is available", () => {
    const component = renderer.create(
      <Table>
        <TablePagination
          colSpan={1}
          hasNextPage={true}
          hasPreviousPage={false}
          onPreviousPage={jest.fn()}
          onNextPage={jest.fn()}
        />
      </Table>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when previous and next pages are available", () => {
    const component = renderer.create(
      <Table>
        <TablePagination
          colSpan={1}
          hasNextPage={true}
          hasPreviousPage={true}
          onPreviousPage={jest.fn()}
          onNextPage={jest.fn()}
        />
      </Table>
    );
    expect(component).toMatchSnapshot();
  });
});
