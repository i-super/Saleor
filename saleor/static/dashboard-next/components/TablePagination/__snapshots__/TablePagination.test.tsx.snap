// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TablePagination /> renders when next page is available 1`] = `
<table
  className="MuiTable-root-1"
>
  <td
    aria-sort={null}
    className="MuiTableCell-root-11 MuiTablePagination-root-5"
    colSpan={1}
    scope={undefined}
  >
    <div
      className="MuiToolbar-root-19 MuiToolbar-gutters-20 MuiTablePagination-toolbar-10"
    >
      <div
        className="MuiTablePagination-spacer-9"
      />
      <div
        className="MuiTablePaginationActions-root-21"
      >
        <button
          className="MuiButtonBase-root-28 MuiButtonBase-disabled-29 MuiIconButton-root-22 MuiIconButton-disabled-26"
          disabled={true}
          onBlur={[Function]}
          onClick={[MockFunction]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="-1"
          type="button"
        >
          <span
            className="MuiIconButton-label-27"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-31"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </span>
        </button>
        <button
          className="MuiButtonBase-root-28 MuiIconButton-root-22"
          disabled={false}
          onBlur={[Function]}
          onClick={[MockFunction]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-27"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-31"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-37"
          />
        </button>
      </div>
    </div>
  </td>
</table>
`;

exports[`<TablePagination /> renders when no previous / next page are available 1`] = `
<table
  className="MuiTable-root-1"
>
  <td
    aria-sort={null}
    className="MuiTableCell-root-11 MuiTablePagination-root-5"
    colSpan={1}
    scope={undefined}
  >
    <div
      className="MuiToolbar-root-19 MuiToolbar-gutters-20 MuiTablePagination-toolbar-10"
    >
      <div
        className="MuiTablePagination-spacer-9"
      />
      <div
        className="MuiTablePaginationActions-root-21"
      >
        <button
          className="MuiButtonBase-root-28 MuiButtonBase-disabled-29 MuiIconButton-root-22 MuiIconButton-disabled-26"
          disabled={true}
          onBlur={[Function]}
          onClick={[MockFunction]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="-1"
          type="button"
        >
          <span
            className="MuiIconButton-label-27"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-31"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </span>
        </button>
        <button
          className="MuiButtonBase-root-28 MuiButtonBase-disabled-29 MuiIconButton-root-22 MuiIconButton-disabled-26"
          disabled={true}
          onBlur={[Function]}
          onClick={[MockFunction]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="-1"
          type="button"
        >
          <span
            className="MuiIconButton-label-27"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-31"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </td>
</table>
`;

exports[`<TablePagination /> renders when previous and next pages are available 1`] = `
<table
  className="MuiTable-root-1"
>
  <td
    aria-sort={null}
    className="MuiTableCell-root-11 MuiTablePagination-root-5"
    colSpan={1}
    scope={undefined}
  >
    <div
      className="MuiToolbar-root-19 MuiToolbar-gutters-20 MuiTablePagination-toolbar-10"
    >
      <div
        className="MuiTablePagination-spacer-9"
      />
      <div
        className="MuiTablePaginationActions-root-21"
      >
        <button
          className="MuiButtonBase-root-28 MuiIconButton-root-22"
          disabled={false}
          onBlur={[Function]}
          onClick={[MockFunction]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-27"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-31"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-37"
          />
        </button>
        <button
          className="MuiButtonBase-root-28 MuiIconButton-root-22"
          disabled={false}
          onBlur={[Function]}
          onClick={[MockFunction]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-27"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-31"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-37"
          />
        </button>
      </div>
    </div>
  </td>
</table>
`;

exports[`<TablePagination /> renders when previous page is available 1`] = `
<table
  className="MuiTable-root-1"
>
  <td
    aria-sort={null}
    className="MuiTableCell-root-11 MuiTablePagination-root-5"
    colSpan={1}
    scope={undefined}
  >
    <div
      className="MuiToolbar-root-19 MuiToolbar-gutters-20 MuiTablePagination-toolbar-10"
    >
      <div
        className="MuiTablePagination-spacer-9"
      />
      <div
        className="MuiTablePaginationActions-root-21"
      >
        <button
          className="MuiButtonBase-root-28 MuiIconButton-root-22"
          disabled={false}
          onBlur={[Function]}
          onClick={[MockFunction]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-27"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-31"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-37"
          />
        </button>
        <button
          className="MuiButtonBase-root-28 MuiButtonBase-disabled-29 MuiIconButton-root-22 MuiIconButton-disabled-26"
          disabled={true}
          onBlur={[Function]}
          onClick={[MockFunction]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="-1"
          type="button"
        >
          <span
            className="MuiIconButton-label-27"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-31"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </td>
</table>
`;
